File: core/src/com/gmaniliapp/flappyhermes/state/State.java
package com.gmaniliapp.flappyhermes.state;

import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.Vector3;

public abstract class State {

    protected OrthographicCamera orthographicCamera;
    protected Vector3 mouse;
    protected GameStateManager gameStateManager;

    protected State(GameStateManager gameStateManager) {
        this.gameStateManager = gameStateManager;
        orthographicCamera = new OrthographicCamera();
        mouse = new Vector3();
    }

    public abstract void update(float deltaTime);

    public abstract void render(SpriteBatch spriteBatch);

    public abstract void dispose();

    protected abstract void handleInput();
}

File: core/src/com/gmaniliapp/flappyhermes/state/State.kt
package com.gmaniliapp.flappyhermes.state

import com.badlogic.gdx.graphics.OrthographicCamera
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.math.Vector3

abstract class State protected constructor(protected var gameStateManager: GameStateManager) {

    protected var orthographicCamera: OrthographicCamera = OrthographicCamera()

    abstract fun update(deltaTime: Float)

    abstract fun render(spriteBatch: SpriteBatch)

    abstract fun dispose()

    protected abstract fun handleInput()

}